src/cdvdcontrol.cpp

```diff
@@ -0,0 +1,1433 @@
+        if (flags_as & FLAS_CREATE) std::cout << " AS_CREATE";
+        if (flags_as & FLAS_DEL)    std::cout << " AS_DEL";
+        if (flags_as & FLAS_VIEW)   std::cout << " AS_VIEW";
+        if (flags_as & FLAS_ACT)    std::cout << " AS_ACT";
+        if (flags_as & FLAS_DEACT)  std::cout << " AS_DEACT";
+        if (flags_as & FLAS_CLEAR)  std::cout << " AS_CLEAR";
+        std::cout << "\n\n";
+    }
+//  std::cout << "____________________________\n";
+    if (flags & FL_LOCK) {
+//      dev->silent++;
+        switch (lock) {
+            case 0:
+                std::cout << "Unlocking media...\n";
+                dev->parms.status &= (~STATUS_LOCK);
+                break;
+            case 1:
+                std::cout << "Locking media...\n";
+                dev->parms.status |= STATUS_LOCK;
+                break;
+            case 2:
+                std::cout << "Toggle media lock state...\n";
+                get_lock(dev);
+                dev->parms.status ^= STATUS_LOCK;
+                break;
+        }
+        set_lock(dev);
+        std::cout << "Media is " << ((dev->parms.status & STATUS_LOCK) ? "" : " NOT") << " locked\n";
+//      dev->silent--;
+    }
+
+    if (flags & FL_LOEJ) {
+//      std::cout << "loej_immed: " << loej_immed << "\n";
+        if (eject == 2) {
+            load_eject(dev, loej_immed);
+        } else {
+            load_eject(dev,!eject, loej_immed);
+        }
+    }
+
+// PLEXTOR features
+    if (flags & FL_POWEREC) {
+//      std::cout << "Set PoweREC...\n";
+        dev->plextor.powerec_state = powerec;
+        plextor_set_powerec(dev);
+    }
+/*
+    if (flags & FL_YMH_AMQR) {
+        dev->yamaha.amqr = amqr;
+        yamaha_set_amqr(dev);
+    }
+    if (flags & FL_YMH_FORCESPEED) {
+        dev->yamaha.forcespeed = forcespeed;
+        yamaha_set_forcespeed(dev);
+    }
+*/
+    if (flags & FL_GIGAREC) {
+//      std::cout << "Set GigaREC...\n";
+        dev->plextor.gigarec = gigarec;
+        plextor_set_gigarec(dev);
+    }
+    if (flags & FL_VARIREC_CD) {
+//      std::cout << "Set VariREC CD...\n";
+//      std::cout << "PWR = %02X   STR = %02X\n",varirec_cd_pwr, varirec_cd_str);
+        dev->plextor.varirec_state_cd = !(flags & FL_VARIREC_CD_OFF);
+        dev->plextor.varirec_pwr_cd = varirec_cd_pwr;
+        dev->plextor.varirec_str_cd = varirec_cd_str;
+        plextor_set_varirec(dev, VARIREC_CD);
+    }
+    if (flags & FL_VARIREC_DVD) {
+//      std::cout << "Set VariREC DVD...\n";
+//      std::cout << "PWR = %02X   STR = %02X\n",varirec_dvd_pwr, varirec_dvd_str);
+        dev->plextor.varirec_state_dvd = !(flags & FL_VARIREC_DVD_OFF);
+        dev->plextor.varirec_pwr_dvd = varirec_dvd_pwr;
+        dev->plextor.varirec_str_dvd = varirec_dvd_str;
+        plextor_set_varirec(dev, VARIREC_DVD);
+    }
+    if (flags & FL_NOSECUREC) {
+        plextor_set_securec(dev, 0, NULL);
+    }
+    else if (flags & FL_SECUREC) {
+        int pwdlen = (char)strlen(passwd);
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)